app/scripts/modules/core/src/delivery/details/StageFailureMessage.tsx

```diff
@@ -37,13 +41,27 @@ export class StageFailureMessage extends React.Component<IStageFailureMessagePro
     if (stage && (stage.isFailed || stage.isStopped)) {
       const failedTask = (stage.tasks || []).find(t => t.status === 'TERMINAL' || t.status === 'STOPPED');
       let failedStage = stage;
-      let failedStageIndex = -1;
-      if (!failedTask && stage.after) {
-        failedStage = stage.after.find(s => s.status === 'TERMINAL' || s.status === 'STOPPED');
-        failedStageIndex = stage.after.indexOf(failedStage);
+      let failedExecutionId;
+      let failedStageIndex;
+      let failedStepIndex;
+
+      if (!failedTask || (!props.message && props.messages.length === 0)) {
+        const exceptionSource: any = get(stage.context, 'exception.source');
+        if (exceptionSource) {
+          failedStage = {
+            id: exceptionSource.stageId,
+            name: exceptionSource.stageName
+          } as IExecutionStage;
```

Comment:
Not crazy about this. `IExecutionStage` has a lot of fields that aren't here (type, context, status, time, tasks, etc). If someone tries to access those fields, we'll end up with a runtime exception.

If these are the only fields we'll care about elsewhere, let's define an interface that declares those fields and use it in the component state.

Reply:
Good call, no need for `IExecutionStage` anyhow. I switched to `failedStageName: string` as this component is only interested in the name anyhow.